.github/workflows/sync-public-with-private.yml

```
@@ -0,0 +1,27 @@
+name: Sync IC private from IC public
+
+on:
+  schedule:
+    - cron: '*/10 * * * *'
+  workflow_dispatch:
+
+permissions:
+  contents: write
+
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        ref: 'master-private'
+
+    - name: Sync changes from public ic
+      run: |
+        git config --global user.name "IDX GitHub Automation"
+        git config --global user.email "IDX GitHub Automation"  
+
+        git remote add public https://github.com/dfinity/ic.git
+        git pull public master -X theirs
+        git push origin master-private
```

What about not running the risk of creating merge commits and just force push public master to master-private:
```suggestion
        git fetch public master
        git push --force origin public/master:master-private
```
(untested)

---

Hmm if I understood the `-X theirs` correctly it should favor the public git history and choose that one.

---

Possibly. Although it's simpler to just skip the merge and only fetch and push. Less risk of things going wrong.

But I'm good either way as long as it works reliably.

---

I took your suggestion but removed `--force`. If everything is working as expected, then master-private should not receive any updates so `--force` should not be necessary. If it does fail, then that's a signal something isn't working as expected.